ui/component/claimTilesDiscover/view.jsx

```diff
@@ -200,30 +233,39 @@ function ClaimTilesDiscover(props: Props) {
 
   return (
     <ul ref={listRef} className="claim-grid">
-      {finalUris && finalUris.length
+      {!loading && finalUris && finalUris.length
         ? finalUris.map((uri, i) => {
             if (uri) {
               const inj = getInjectedItem(i);
-              return (
-                <React.Fragment key={uri}>
-                  {inj && inj}
-                  {(!inj || !injectedItem || !injectedItem.replace) && (
+              if (inj && adBlockerFound && !hasPremiumPlus) {
+                return (
+                  <React.Fragment key={uri}>
+                    <PremiumPlus />
+                  </React.Fragment>
+                );
+              } else if (inj) {
+                return <React.Fragment key={uri}>{inj}</React.Fragment>;
+              } else {
+                return (
+                  <React.Fragment key={uri}>
```

Comment:
Okay, makes sense on the channel pages, but injections are the reason why the current grid is broken. Who came up with this idea? It's always gonna replace an element on the individual level instead of the grid level which makes it impossible to adjust the array without constant communication between the parent and the child.

Reply:
Agree. My second bullet point could be a misunderstanding, so can ignore.  Thought process was:
- "replace" was the original implementation in Homepage, then was asked to do "insert" because a user complaint about a missing tile from creator in the homepage, which I didn't understand since it's a truncated list and there is a "Show more" button to see everything.  Then, I thought the next request is to continue with "insert" but fill remaining space, which I didn't know how since sometimes there is no more data to fill with.